prow/config/tide.go

```diff
@@ -172,3 +173,36 @@ func (tqs TideQueries) QueryMap() QueryMap {
 func (q QueryMap) ForRepo(org, repo string) TideQueries {
 	return append(append(TideQueries(nil), q[org]...), q[fmt.Sprintf("%s/%s", org, repo)]...)
 }
+
+func (tq *TideQuery) Validate() error {
+	for o := range tq.Orgs {
+		if strings.Contains(tq.Orgs[o], "/") {
+			return fmt.Errorf("orgs[%d]: %q contains a '/' which is not valid", o, tq.Orgs[o])
+		}
+		if len(tq.Orgs[o]) == 0 {
+			return fmt.Errorf("orgs[%d]: is an empty string", o)
+		}
+	}
+
+	for r := range tq.Repos {
+		parts := strings.Split(tq.Repos[r], "/")
+		if len(parts) != 2 || len(parts[0]) == 0 || len(parts[1]) == 0 {
+			return fmt.Errorf("repos[%d]: %q is not of the form \"org/repo\"", r, tq.Repos[r])
+		}
+		for o := range tq.Orgs {
+			if tq.Orgs[o] == parts[0] {
+				return fmt.Errorf("repos[%d]: %q is already included via orgs[%d]: %q", r, tq.Repos[r], o, tq.Orgs[o])
+			}
+		}
+	}
+
+	if len(tq.ExcludedBranches) > 0 && len(tq.IncludedBranches) > 0 {
+		return errors.New("cannot supply both included and excluded branches")
```

Comment:
Why? Isn't this handled fine by the query?

Reply:
I was thinking that we should consider this an error because it never makes sense to specify both a whitelist and a blacklist so it is presumably a misconfiguration. It isn't technically wrong, but the blacklist doesn't matter if there is a whitelist.
I'm happy to remove this if you'd like to allow both to be specified. It just seemed like a smell that we could catch.

Reply:
If you have an existing blacklist and don't want to touch it when you know you're turning on a white list for only a short period, this would be helpful. The idea for the whilelists was for transient config update, right? I see the point abouta smell but maybe it would be nice to do minimal config updates? This is a nit either way though

Reply:
I don't see how a whitelist would make it any easier to make temporary config changes.

I'll remove this check though. I suppose we shouldn't error on smelly (but valid) GitHub search syntax in case someone does have a reason to do something smelly.

Reply:
We can at least emit a warning

Reply:
Done.